sdks/java/harness/src/test/java/org/apache/beam/fn/harness/state/FakeBeamFnStateClient.java

```diff
@@ -47,6 +60,7 @@ public class FakeBeamFnStateClient implements BeamFnStateClient {
   private static final int DEFAULT_CHUNK_SIZE = 6;
   private final Map<StateKey, List<ByteString>> data;
   private int currentId;
+  private final Map<StateKey, NavigableSet<Long>> orderedListSortKeysFromStateKey;
```

Comment:
This seems like a pretty inefficient data structure to use. You might look at `SortedSet` (which requires adding a deduplicating id) unless we can use `SortedMultiset` from Guava. This would allow us to do things like remove all elements in a range without first copying over all keys in the range and then removing them one by one (which requires a hash each time).

Reply:
Oh, I see this is part of the "fake" client, so maybe we don't care as much about performance?

Reply:
The structure is used to support multiple ordered list states within one DoFn. 

For example, we can have order list state A and B, which have different StateKeys. The outer Map is only used once when looking up the inner set structure.

Then under each state key, the NavigableSet, which is internally initialized with a TreeSet, can support removing a range of keys efficiently.